pkg/identityserver/store/secrets.go

```diff
@@ -0,0 +1,39 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import "go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+
+// Secrets model.
+type Secrets struct {
+	Model
+	SoftDelete
+
+	Data []byte `gorm:"type:BYTEA"`
```

Comment:
This doesn't feel right. I think it's better to store entity in the same way we store attributes, so something like

```go
type Secret struct {
	ID string `gorm:"type:UUID;primary_key;default:gen_random_uuid()"`

	EntityID   string `gorm:"type:UUID;index:attribute_entity_index;not null"`
	EntityType string `gorm:"type:VARCHAR(32);index:attribute_entity_index;not null"`

	Key             string `gorm:"type:VARCHAR"`
	EncryptionKeyID string `gorm:"type:VARCHAR"`
	Ciphertext      string `gorm:"type:BYTEA"`
}
```